message.go

```diff
@@ -465,8 +462,9 @@ func (ssa *SingleSegmentArena) Allocate(sz Size, segs map[SegmentID]*Segment) (S
 	if err != nil {
 		return 0, nil, err
 	}
-	buf := make([]byte, len(data), cap(data)+inc)
-	copy(buf, data)
+	buf := bufferpool.Default.Get(cap(data) + inc)
+	buf = buf[:copy(buf, data)]
```

Comment:
Shoving all this on one line makes it hard to follow imo; maybe:

```
copied := copy(buf, data)
buf = buf[:copied]
```

Reply:
This shouldn't have been pushed to this PR.  I've corrected it in #465.